timbermill-java/timbermill-local/src/main/java/com/datorama/oss/timbermill/ElasticsearchClient.java

```diff
@@ -948,5 +951,113 @@ public Bulker getBulker() {
 	public IndexRetryManager getRetryManager() {
 		return retryManager;
 	}
+
+    public void mergeIndexes() {
+		Set<String> currentAliases = getCurrentAliases();
+		currentAliases.forEach(alias -> {
+			List<String> indicesToBeMoved = getIndicesToBeMoved(alias);
+			List<List<String>> partition = Lists.partition(indicesToBeMoved, 2);
+			partition.forEach(indexes -> {
+				if (indexes.size() <= 1) {
+					return;
+				}
+				String sourceIndex = indexes.get(0);
+				String destIndex = indexes.get(1);
+				moveIndexAnotherIndex(sourceIndex, destIndex);
+			});
+		});
+    }
+
+	private Set<String> getCurrentAliases() {
+		return getAliases(TIMBERMILL_INDEX_WILDCARD).keySet().stream().map(index -> index.replaceAll(INDEX_DELIMITER + "\\d+$", "")).collect(Collectors.toSet());
+	}
+
+    private List<String> getIndicesToBeMoved(String alais) {
+        try {
+			Set<String> indexNotToMerge = getAliases(alais + "*").keySet();
+            Request request = new Request("GET", "/_cat/indices/" + alais + "*?format=json");
+            Response response = client.getLowLevelClient().performRequest(request);
+            InputStream content = response.getEntity().getContent();
+            String json = IOUtils.toString(content);
+            JsonArray asJsonObject = new JsonParser().parse(json).getAsJsonArray();
+            List<String> indicesToMove = new ArrayList<>();
+            asJsonObject.forEach(jsonElement -> {
+                JsonObject jsonObject = jsonElement.getAsJsonObject();
+                String storeSize = jsonObject.get("store.size").getAsString();
+                String index = jsonObject.get("index").getAsString();
+                if (indexNotToMerge.contains(index)) {
+                    return;
+                }
+                if (isSizeLowerThen(storeSize, (long) (maxIndexSizeInGB * 0.1))) { // 10% of max index size
+                    indicesToMove.add(index);
+                }
+            });
+
+            LOG.info("Number of indices to be moved: {}", indicesToMove.size());
```

Comment:
Lets also print the names of the indices and their sizes that we decided to move

Reply:
done but in a different place